Silence "Duplicate specification" warnings - better handling

tyboro opened the pull request:
more explicit way to silence 'Duplicate specification' warnings
also, on my Windows machine, I no longer get the following:
```
C:\Users\mark>pg_prove --help
Duplicate specification "psql-bin|b=s" for option "b"
Duplicate specification "host|h=s" for option "h"
Duplicate specification "port|p=s" for option "p"
Duplicate specification "pset|P=s%" for option "P"
Duplicate specification "schema|s=s" for option "s"
Duplicate specification "version|V" for option "version"
Duplicate specification "version|V" for option "V"
Duplicate specification "help|H|?" for option "help"
Duplicate specification "help|H|?" for option "H"
Duplicate specification "help|H|?" for option "?"
Duplicate specification "man|m" for option "man"
Duplicate specification "man|m" for option "m"
Usage:
      pg_prove tests/
      pg_prove --dbname template1 test*.sql
      pg_prove -d testdb --runtests
...
```

theory commented:
Looks good, but why on earth are you getting those warnings? Seems odd.

tyboro commented:
yeah, I was puzzled by the warnings...
but then, I got annoyed just enough to find a way to suppress them ;-)
btw, I run Windows 10, with a fresh Strawberry install

jbeck111 commented:
I also get the warnings, and would love not to...

bash-5.2$ pg_prove
Duplicate specification "psql-bin|b=s" for option "b"
Duplicate specification "host|h=s" for option "h"
Duplicate specification "port|p=s" for option "p"
Duplicate specification "pset|P=s%" for option "P"
Duplicate specification "schema|s=s" for option "s"
Duplicate specification "version|V" for option "version"
Duplicate specification "version|V" for option "V"
Duplicate specification "help|H|?" for option "help"
Duplicate specification "help|H|?" for option "H"
Duplicate specification "help|H|?" for option "?"
Duplicate specification "man|m" for option "man"
Duplicate specification "man|m" for option "m"
Files=0, Tests=0,  0 wallclock secs ( 0.01 usr +  0.00 sys =  0.01 CPU)
Result: NOTESTS

theory commented:
I noticed them myself last week. Will dig into it.

Commit:
Silence "Duplicate specification" warnings

Override `$SIG{__WARN__}` since overriding `$^W` is not (no longer?)
sufficient.
